packages/connectivity/src/http-agent/http-agent.ts

```diff
@@ -14,28 +14,49 @@ import {
   proxyAgent,
   proxyStrategy
 } from '../scp-cf/destination';
+import { registerDestinationCache } from '../scp-cf/destination/register-destination-cache';
 import { HttpAgentConfig, HttpsAgentConfig } from './agent-config';
 
 const logger = createLogger({
   package: 'connectivity',
   messageContext: 'http-agent'
 });
 
+/**
+ * Will be renamed to getAgentConfig in the next major release.
+ * Returns a promise of the http or https-agent config depending on the destination URL.
+ * If the destination contains a proxy configuration, the agent will be a proxy-agent.
+ * If not it will be the default http-agent coming from node.
+ * @param destination - Determining which kind of configuration is returned.
+ * @returns A promise of the HTTP or HTTPS agent configuration.
+ */
+export async function getAgentConfigAsync(
+  destination: HttpDestination
+): Promise<HttpAgentConfig | HttpsAgentConfig> {
+  const certificateOptions = {
+    ...getTrustStoreOptions(destination),
+    ...getKeyStoreOption(destination),
+    ...(await getMtlsOptions(destination))
+  };
+  return destination.proxyConfiguration
+    ? proxyAgent(destination, certificateOptions)
+    : createDefaultAgent(destination, certificateOptions);
+}
+
 /**
  * Returns the http or https-agent config depending on the destination URL.
  * If the destination contains a proxy configuration, the agent will be a proxy-agent.
  * If not it will be the default http-agent coming from node.
+ * @deprecated Temporarily replaced by {@link getAgentConfigAsync}, will change its default behavior to be asynchronous in next major release.
```

Comment:
[q] Do you think it makes sense to still add the mtls options here as a synchronous variant?

Reply:
I'd rather not, it adds more maintenance overhead and is worse than the async function, since we can already remove it (as it was tagged with @experimental) I would cut it out ASAP